src/go/k8s/pkg/resources/statefulset.go

```diff
@@ -56,20 +56,21 @@ const (
 // focusing on the management of redpanda cluster
 type StatefulSetResource struct {
 	k8sclient.Client
-	scheme                      *runtime.Scheme
-	pandaCluster                *redpandav1alpha1.Cluster
-	serviceFQDN                 string
-	serviceName                 string
-	nodePortName                types.NamespacedName
-	nodePortSvc                 corev1.Service
-	redpandaCertSecretKey       types.NamespacedName
-	internalClientCertSecretKey types.NamespacedName
-	adminCertSecretKey          types.NamespacedName
-	adminAPINodeCertSecretKey   types.NamespacedName
-	adminAPIClientCertSecretKey types.NamespacedName
-	serviceAccountName          string
-	configuratorTag             string
-	logger                      logr.Logger
+	scheme                         *runtime.Scheme
+	pandaCluster                   *redpandav1alpha1.Cluster
+	serviceFQDN                    string
+	serviceName                    string
+	nodePortName                   types.NamespacedName
+	nodePortSvc                    corev1.Service
+	redpandaCertSecretKey          types.NamespacedName
+	internalClientCertSecretKey    types.NamespacedName
+	adminCertSecretKey             types.NamespacedName // TODO this is unused, can be removed
```

Comment:
yeah I think this gets also addressed in my PR, I'll get back to it when this merges